### DISCUSSION TITLE
PETSc problems

### DISCUSSION THREAD

**Discussion by adamclarence** [wenceorg/comp4187#28] [Q&A] [answered]
Hello,

I've (finally!) managed to get WSL working and running Ubuntu, and I have tried installing `petsc4py`. I had the same error message as [Oliver's](https://github.com/wenceorg/comp4187/discussions/24#discussioncomment-464623) when I just ran `pip install petsc4py`, so I tried
`PETSC_CONFIGURE_OPTIONS=--download-fblaslapack=1 pip install petsc`
as you said, but it is still not working.

In particular, the error message is:
`    *******************************************************************************
             UNABLE to CONFIGURE with GIVEN OPTIONS    (see configure.log for details):
    -------------------------------------------------------------------------------
    Cannot use fblaslapack without Fortran, make sure you do NOT have --with-fc=0
    *******************************************************************************`

I have tried setting different options for `--with-fc`, but these all yield the error `RuntimeError: Do not use --with-fc, use the environmental variable MPIF90`, and I don't know what to do from here. Do you know what's going wrong?

**Comment by wence-** [comment, marked as answer]
I suspect that you do not have a useable fortran compiler on the system. Debugging this via the pip install process is a little fiddly, so let's build and configure petsc separately and then use it from petsc4py.

If you get the tarball from here:

https://ftp.mcs.anl.gov/pub/petsc/release-snapshots/petsc-3.14.5.tar.gz

And unpack it, it uses a `./configure; make` build process. Run `./configure --download-fblaspack=1` and petsc will churn away for a while. Eventually, I suspect it will fail, but you'll have access to `configure.log` which will contain much more information.

**Reply by wence-** [reply]
Actually, given that you run ubuntu inside WSL let's make sure that we have a dev environment available with compilers. I think you probably want to `apt install` the following packages

```
build-essential autoconf automake gfortran python3-dev python3-pip python3-venv
```

before running the petsc configure.

**Reply by wence-** [reply]
Did this work?

**Reply by adamclarence** [reply]
No errors yet, but it's still running

**Reply by adamclarence** [reply]
Yes, that has worked. Thank you very much!
